unique_toolkit/unique_toolkit/language_model/service.py

```diff
@@ -0,0 +1,223 @@
+
+import unique_sdk
+
+from unique_toolkit.chat.state import ChatState
+from unique_toolkit.content.schemas import ContentChunk
+from unique_toolkit.language_model.infos import LanguageModelName
+from unique_toolkit.language_model.schemas import (
+    LanguageModelMessages,
+    LanguageModelResponse,
+    LanguageModelStreamResponse,
+)
+from unique_toolkit.performance.async_wrapper import async_warning, to_async
+
+
+class LanguageModelService:
+    def __init__(self, state: ChatState, logger: Optional[logging.Logger] = None):
+        self.state = state
+        self.logger = logger or logging.getLogger(__name__)
+
+    _DEFAULT_COMPLETE_TIMEOUT = 240000
+    _DEFAULT_COMPLETE_STREAM_TIMEOUT = 100000
+    _DEFAULT_COMPLETE_TEMPERATURE = 0.0
+    _DEFAULT_COMPLETE_STREAM_TEMPERATURE = 0.25
+
+    def complete(
+        self,
+        messages: LanguageModelMessages,
+        model_name: LanguageModelName,
+        temperature: float = _DEFAULT_COMPLETE_TEMPERATURE,
+        timeout: int = _DEFAULT_COMPLETE_TIMEOUT,
+    ):
+        """
+        Calls the completion endpoint synchronously without streaming the response.
+
+        Args:
+            messages (LanguageModelMessages): The LanguageModelMessages obj to complete.
+            model_name (LanguageModelName): The model name.
+            temperature (float): The temperature value. Defaults to 0.
+            timeout (int): The timeout value in milliseconds. Defaults to 240000.
+
+        Returns:
+            LanguageModelResponse: The LanguageModelResponse object.
+        """
+        return self._trigger_complete(
+            messages=messages,
+            model_name=model_name,
+            temperature=temperature,
+            timeout=timeout,
+        )
+
+    @to_async
+    @async_warning
+    def async_complete(
+        self,
+        messages: LanguageModelMessages,
+        model_name: LanguageModelName,
+        temperature: float = _DEFAULT_COMPLETE_TEMPERATURE,
+        timeout: int = _DEFAULT_COMPLETE_TIMEOUT,
+    ):
+        """
+        Calls the completion endpoint asynchronously without streaming the response.
+
+        Args:
+            messages (LanguageModelMessages): The messages to complete.
+            model_name (LanguageModelName): The model name.
+            temperature (float): The temperature value. Defaults to 0.
+            timeout (int): The timeout value in milliseconds. Defaults to 240000.
+
+        Returns:
+            str: The completed message content.
+        """
+        return self._trigger_complete(
+            messages=messages,
+            model_name=model_name,
+            temperature=temperature,
+            timeout=timeout,
+        )
+
+    def _trigger_complete(
+        self,
+        messages: LanguageModelMessages,
+        model_name: LanguageModelName,
+        temperature: float,
+        timeout: int,
+    ) -> LanguageModelResponse:
+        messages = messages.model_dump(exclude_none=True)
+
```

Comment:
Bring tools into openai format. Not sure if we already have such a function (will see when I reach the tool section)

```suggestion
       options = {}
        if tools is not None and len(tools):
            options["tools"] = [
                self.tool_service.to_openai_format(tool) for tool in tools
            ]
```

Reply:
Here the function I used here:

```
def to_openai_format(self, tool: Tool) -> dict:
        return {"type": "function", "function": tool.model_dump(exclude_none=True)}
```